sys/include/net/nanocoap_sock.h

```diff
@@ -229,9 +229,7 @@ typedef struct {
  */
 typedef struct {
     sock_udp_ep_t remote;           /**< remote to send response to         */
-#if defined(MODULE_SOCK_AUX_LOCAL) || DOXYGEN
     sock_udp_ep_t local;            /**< local from which to send response  */
-#endif
```

Comment:
Because it will reply from a random port without `sock_aux_local` - which will render it a random unexpected UDP packet rather than a reply. That is why `sock_aux_local` has been added as dependency of `nanocoap_server_separate`.

In Wireshark it even is not labeled as CoAP without this.

Reply:
Ah I didn't see that this is `nanocoap_server_response_ctx_t` which is of course only used with `nanocoap_server_separate`